### ISSUE TITLE
sidekiq-pro suddenly stopping and will not restart

### ISSUE THREAD

**Issue by DaveCollinsJr** [mperham/sidekiq#2665]
Hello,

  Today, after a routine update of our Rails application, sidekiq-pro on this server seems to die after processing some jobs and then will not restart.

Our Versions:
/var/sentrylink/shared/bundle/ruby/2.0.0/gems/sidekiq-pro-1.9.2
/var/sentrylink/shared/bundle/ruby/2.0.0/gems/sidekiq-3.3.2

The last thing visible in **/var/log/upstart/sidekiq-manager.log** is:
```
start: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
```

Here are some of our other key files (to my knowledge none of these changed during this update):

**/etc/init/sidekiq.conf**
```
# /etc/init/sidekiq.conf - Sidekiq config

# This example config should work with Ubuntu 12.04+.  It
# allows you to manage multiple Sidekiq instances with
# Upstart, Ubuntu's native service management tool.
#
# See sidekiq-manager.conf for how to manage all Sidekiq instances at once.
#
# Save this config as /etc/init/sidekiq.conf then manage sidekiq with:
#   sudo start sidekiq app=/path/to/app index=0
#   sudo stop sidekiq app=/path/to/app  index=0
#   sudo status sidekiq app=/path/to/app index=0
#
# or use the service command:
#   sudo service sidekiq {start,stop,restart,status}
#

description "Sidekiq Background Worker"

# no "start on", we don't want to automatically start
stop on (stopping sidekiq-manager or runlevel [06])

# change apps to match your deployment user if you want to use this as a less privileged user (recommended!)
setuid vagrant
setgid vagrant
env HOME=/home/vagrant

respawn
respawn limit 3 30

# TERM is sent by sidekiqctl when stopping sidekiq.  Without declaring these as normal exit codes, it just respawns.
normal exit 0 TERM

instance ${app}-${index}

script
# this script runs in /bin/sh by default
# respawn as bash so we can source in rbenv
exec /bin/bash <<'EOT'
  # Pick your poison :) Or none if you're using a system wide installed Ruby.
  # rbenv
  # source /home/apps/.bash_profile
  # OR
  # source /home/apps/.profile
  # OR system:
  # source /etc/profile.d/rbenv.sh
  #
  # rvm
  source /home/vagrant/.rvm/bin/rvm

  logger -t sidekiq "Starting process: $app-$index"

  # Logs out to /var/log/upstart/sidekiq.log by default

  cd $app
  exec bundle exec sidekiq -i ${index} -e production -P ${app}/tmp/pids/sidekiq-${index}.pid
EOT
end script

pre-stop script
# this script runs in /bin/sh by default
# respawn as bash so we can source in rbenv
exec /bin/bash <<'EOT'
  # Pick your poison :) Or none if you're using a system wide installed Ruby.
  # rbenv
  # source /home/apps/.bash_profile
  # OR
  # source /home/apps/.profile
  # OR system:
  # source /etc/profile.d/rbenv.sh
  #
  # rvm
  # source /home/apps/.rvm/scripts/rvm

  logger -t sidekiq "Stopping process: $app-$index"

  # Logs out to /var/log/upstart/sidekiq.log by default

  cd $app
  exec bundle exec sidekiqctl stop ${app}/tmp/pids/sidekiq-${index}.pid 2> /dev/null
EOT
end script
```

**/etc/init/sidekiq-manager.conf**
```
# /etc/init/sidekiq-manager.conf - manage a set of Sidekiqs

# This example config should work with Ubuntu 12.04+.  It
# allows you to manage multiple Sidekiq instances with
# Upstart, Ubuntu's native service management tool.
#
# See sidekiq.conf for how to manage a single Sidekiq instance.
#
# Use "stop sidekiq-manager" to stop all Sidekiq instances.
# Use "start sidekiq-manager" to start all instances.
# Use "restart sidekiq-manager" to restart all instances.
# Crazy, right?
#

description "Manages the set of sidekiq processes"

# This starts upon bootup and stops on shutdown
start on runlevel [2345]
stop on runlevel [06]

# Set this to the number of Sidekiq processes you want
# to run on this machine
env SIDEKIQ_CONF=/etc/sidekiq.conf

pre-start script
  for i in `cat $SIDEKIQ_CONF`; do
    app=`echo $i | cut -d , -f 1`
    num_workers=`echo $i | cut -d , -f 2`
    for j in `seq 0 $(($num_workers - 1))`; do
      start sidekiq app=$app index=$j
    done
  done
end script
```

**/etc/sidekiq.conf**
```
/var/sentrylink/current,3
```

Any help is greatly appreciated!  We are currently down in this production environment.

**Comment by mperham** [comment]
None of that info gives me any clue. Any other log file output?

**Comment by DaveCollinsJr** [comment]
You are not going to believe this one: :rage1: 
Wasn't it Shakespeare who said: *"The first thing we do, let's kill all the lawyers"*??

Turns out that, at the advice of an attorney, someone had put a copyright notice at the top of EVERY file including /etc/sidekiq.conf... Of course sidekiq was trying to parse (c) and hilarity ensued.

My clue was seeing this in the log as I looked further back:
```
/proc/self/fd/9: 1: /proc/self/fd/9: arithmetic expression: expecting primary: "# - 1"
/proc/self/fd/9: 1: /proc/self/fd/9: arithmetic expression: expecting primary: "© - 1"
```

Is resolved now.  Sorry for wasting your time!
